onboarding/tests/tests.py

```diff
@@ -0,0 +1,141 @@
+import requests
+
+from django.test import TestCase
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from student.models import UserProfile, Registration
+from onboarding.models import UserExtendedProfile, EmailPreference, Organization
+
+
+class UserDeletionTestCase(TestCase):
+    """
+    Tests whether all of user data is deleted to make the app GDPR compliant
+    """
+    username = "testuser"
+    email = "test@email.com"
+
+    def setUp(self):
+        user = User(
+            username=self.username,
+            email=self.email,
+        )
+        user.set_password("Arbisoft2")
+        user.save()
+        user.first_name = "Test2"
+        user.save()
+
+        user_data = {
+            'edx_user_id': user.id,
+            'email': user.email,
+            'first_name': "Test",
+            'last_name': "User",
+            'username': user.username,
+            'date_joined': user.date_joined.strftime('%d/%m/%Y'),
+        }
+
+        email_preference = EmailPreference(
+            user=user,
+            opt_in="no",
+
+        )
+        email_preference.save()
+
+        user_profile = UserProfile(
+            user=user,
+            name="Test User",
+            level_of_education='b',
+        )
+        user_profile.save()
+        user_profile.name = "Test User 2"
+        user_profile.save()
+
+        extended_profile = UserExtendedProfile(
+            user=user,
+            is_interests_data_submitted=True,
+            english_proficiency="Master",
+        )
+        extended_profile.save()
+        extended_profile.english_proficiency = 'Novice',
+        extended_profile.save()
+
+        organization = Organization(
+            admin=user,
+            alternate_admin_email=self.email,
+        )
+        organization.save()
+        organization.unclaimed_org_admin_email = self.email
+        organization.alternate_admin_email = None
+        organization.save()
+
+    def test_user_data_is_deleted(self):
+        """
+        Tests whether all the relevant user data is created
+        """
+        user = User.objects.filter(username=self.username).first()
+        email_preference = EmailPreference.objects.filter(user=user).first()
+        user_profile = UserProfile.objects.filter(user=user).first()
+        extended_profile = UserExtendedProfile.objects.filter(
+            user=user).first()
+
+        data_endpoint = settings.NODEBB_ENDPOINT + '/api/v2/users/data'
+        headers = {'Authorization': 'Bearer ' + settings.NODEBB_MASTER_TOKEN}
```

Comment:
We don't have NODEBB and its settings on the project. This should be refactored.

Did you run tests?

Reply:
Yes, tests don't work. I just removed these tests for now.